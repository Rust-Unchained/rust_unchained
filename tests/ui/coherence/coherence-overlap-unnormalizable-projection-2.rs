pub trait WhereBound {}
impl WhereBound for () {}

pub trait WithAssoc<'a> {
    type Assoc;
}

pub trait Trait {}
impl<T> Trait for T
where
    T: 'static,
    for<'a> T: WithAssoc<'a>,
    for<'a> Box<  <T as WithAssoc<'a>>::Assoc  >: WhereBound,
{
}

impl<T> Trait for Box<T> {}
//~^ ERROR E0119

struct Local;
impl WithAssoc<'_> for Box<Local> {
    type Assoc = Local;
}

impl WhereBound for Box<Local> {}

fn main() {}