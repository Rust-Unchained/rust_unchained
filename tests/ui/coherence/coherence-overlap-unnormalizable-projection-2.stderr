error[E0119]: conflicting implementations of trait `Trait` for type `Box<Local>`
  --> $DIR/coherence-overlap-unnormalizable-projection-2.rs:17:1
   |
LL | / impl<T> Trait for T
LL | | where
LL | |     T: 'static,
LL | |     for<'a> T: WithAssoc<'a>,
LL | |     for<'a> Box<  <T as WithAssoc<'a>>::Assoc  >: WhereBound,
   | |_____________________________________________________________- first implementation here
...
LL |   impl<T> Trait for Box<T> {}
   |   ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Box<Local>`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
